### ISSUE TITLE
Unable to auth via Domain/WAN in app, fine in chrome

### ISSUE THREAD

**Issue by nerdfoundrygaming** [home-assistant/home-assistant-android#239]
**Home Assistant Android version:** 1.3.1
**Android version:** 10
**Phone model:** OnePlus 6
**Home Assistant version:** 0.103.6
**Last working Home Assistant release (if known):** N/A

----

**Description of problem:**

Install Method is Docker on Debian-based host. 
IPv6 is disabled and ensured on entire net stack (router/host/docker/container/nginx/letsencrypt).

**ALL ATTEMPTS ARE MADE VIA A REMOTE CONNECTION** - *I'm currently not concerned about LAN access (as it is working perfectly fine), simply remote/WAN.*

As I've moved to a DuckDNS/LetsEncrypt implementation, I realized the app was still connecting via my LAN IP (#238). In an attempt to convert to the domain, I recieve an error indicating a failure to resolve a url, and an invalid auth attempt.

Also a **VERY** important oddity to note is that I can log in just fine via chrome on my phone, the app simply fails.

I've tried adding trusted networks as follows to enforce auth to no avail (which you can see working fine via Chrome, verified coming from the same internal docker IP via letsencrypt/nginx container):

```
auth_providers:
  - type: homeassistant
  - type: trusted_networks
    trusted_networks:
      - 192.168.1.0/24
      - 172.0.0.0/8
```

I can `exec` into the container(s) (both letsencrypt/nginx and HA) and *can* ping and wget the URI: `http://home-assistant.io/android`, which redirects to https/443 as expected, and retrieves index page of `https://companion.home-assistant.io/` as also expected with follow redirects enabled.

**Traceback (if applicable):**

Technically log, not traces

```
home-assistant    | 2020-01-08 13:59:29 ERROR (MainThread) [homeassistant.components.auth.indieauth] Timeout while looking up redirect_uri https://home-assistant.io/android
home-assistant    | 2020-01-08 13:59:29 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.18.0.2
```

**Additional information:**

`172.18.0.2` is the LE/nginx container IP.

Image album as proof of working in chrome, but not in the app: https://imgur.com/a/M5XhiR4

As you can see, the HA app does appear to acknowledge there are >1 providers available for auth, however I cannot attempt either as the app fails with the red error displayed.

The error in the HA log is confusing, because no such attempt at auth has been made - I can't even get that far in the process. This is with and without `trusted_networks` provider, so without I expect to at least be prompted for a user before an error such as "auth failure" is presented in the log.

**Comment by JBassett** [comment]
Please include the entire build number of the app.  Are you using the google play store version or the firebase version?  At this time I am unable to reproduce on the latest builds.  If you try the latest version from here: https://appdistribution.firebase.dev/i/8zf5W4zz and still see the issue feel free to reopen.

**Comment by nerdfoundrygaming** [comment]
How do I get the full version? The version number given was what Android App Manager tells me it is, because I have no visibility into the app other than a failing login screen.

I installed the firebase version you linked, and still getting the same issue:

```
2020-01-10 10:29:13 ERROR (MainThread) [homeassistant.components.auth.indieauth] Timeout while looking up redirect_uri https://home-assistant.io/android
2020-01-10 10:29:13 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 192.168.1.10
```

Notice this time I'm attempting on my LAN and it's failing in similar fashion. I have a script running every 1s to ping and wget the offending URL (from within the HA container) to see if I am somehow losing connection, but never has it failed. 

I'm not sure what else to check/try. Any suggestions?

**Comment by nerdfoundrygaming** [comment]
So with some help from the awesome people in the Discord, I've been able to determine a few things:

1) Android App Manager doesn't give the full info... Sad-face.
2) The only way to read the version is to have logged in. I think maybe it'd be nice to have the version number in a footer on the login page for communication errors/reports. There wasn't any way for me to give you the full version otherwise.
3) I made a rookie mistake of not restarting my container during this whole testing phase. Upon restarting the container, all is good.

Would it be worth re-testing on upstream, or just live life on the firebase edge 🤔
